libs/components/datetime/src/lib/modules/datepicker/datepicker-input-fuzzy.directive.ts

```diff
@@ -604,4 +533,73 @@ export class SkyFuzzyDatepickerInputDirective
   private onTouched = () => {};
   // eslint-disable-next-line @typescript-eslint/no-empty-function
   private onValidatorChange = () => {};
+
+  private updateValue(value: any, isProgramatic = false): void {
+    if (this._value === value) {
+      return;
+    }
+
+    let fuzzyDate: SkyFuzzyDate;
+    let fuzzyMoment: any;
+    let dateValue: Date;
+    let formattedDate: string;
+
+    if (value instanceof Date) {
+      dateValue = value;
+      formattedDate = this.dateFormatter.format(value, this.dateFormat);
+      fuzzyDate = this.fuzzyDateService.getFuzzyDateFromSelectedDate(
+        value,
+        this.dateFormat
+      );
+    } else if (typeof value === 'string') {
+      fuzzyDate = this.fuzzyDateService.getFuzzyDateFromString(
+        value,
+        this.dateFormat
+      );
+      formattedDate = this.fuzzyDateService.format(
+        fuzzyDate,
+        this.dateFormat,
+        this.locale
+      );
+
+      if (!formattedDate) {
+        formattedDate = value;
+      }
+
+      fuzzyMoment = this.fuzzyDateService.getMomentFromFuzzyDate(fuzzyDate);
+
+      if (fuzzyMoment) {
+        dateValue = fuzzyMoment.toDate();
+      }
+    } else {
+      fuzzyDate = value as SkyFuzzyDate;
+      formattedDate = this.fuzzyDateService.format(
+        fuzzyDate,
+        this.dateFormat,
+        this.locale
+      );
+      fuzzyMoment = this.fuzzyDateService.getMomentFromFuzzyDate(fuzzyDate);
+
+      if (fuzzyMoment) {
+        dateValue = fuzzyMoment.toDate();
+      }
+    }
+
+    const areFuzzyDatesEqual = this.fuzzyDatesEqual(this._value, fuzzyDate);
+    const isNewValue = fuzzyDate !== this._value || !areFuzzyDatesEqual;
+
+    this._value = fuzzyDate || value;
+
+    if (isNewValue) {
+      if (isProgramatic) {
+        this.control?.setValue(this._value, { emitEvent: false });
+      } else {
+        this.onChange(this._value);
+      }
+
+      this.datepickerComponent.selectedDate = dateValue;
+    }
+
+    this.setInputElementValue(formattedDate || '');
```

Comment:
☝️ Are all these changes a simple copy/paste from the original setter? Any changes that I need to focus on?

Reply:
Just a copy and paste of the original setter other than the new if statement for `isProgrammatic` where we set the controls value without using `onChange`.